### ISSUE TITLE
How to train on batches with one huge .tsv dataset

### ISSUE THREAD

**Issue by adrie84** [keras-team/keras#14264]
Hi everybody,

I am new to Keras and ML in general, so I apologize if my question seems naive.

I am trying to train a network on a very large .tsv dataset (about 30GB). The data are a timeseries of spatial coordinates.
My approach is to take a fairly large batch, preprocess it, shuffle it in every dimension, and then divide it in a training and validation set.

The examples/tutorials I found, always use a separate batch generator for training and validation. However, in this case it would be more convenient to generate both training and validation data with one batch generator. Is there an efficient way to do this? 

Thanks for your answer!
